### ISSUE TITLE
dbc.Spinner makes page reload when triggering a callback

### ISSUE THREAD

**Issue by melissadosm** [facultyai/dash-bootstrap-components#463]
Please fill out the below information as much as possible.

- dash version: `#1.14.0`
- dash-bootstrap-components version: `#0.10.3`
- components affected by bug: callbacks

### What is happening?

I'm working with a multi-page dash app. I wrapped my main Div with a dbc.Spinner, and some of the pages on the Div have callbacks. Whenever I triggered a callback, the page reloaded the entire layout (the callback behaves correctly), and I go back to the beginning of the page. I guess what is happening is that whenever the callback triggers dash understands that it has to show the spinner.

### What should be happening?

The entire page should not reload when I trigger a callback. 

### Code

I can't provide an entire working code as it's a multipage app but I hope this gives you a sense of what was happening.

This is my main script

```
import dash
import dash_bootstrap_components as dbc
import dash_core_components as dcc
import dash_html_components as html
from dash.dependencies import Input, Output

from importlib import import_module


app = dash.Dash(
    __name__, 
    meta_tags=[{"name": "viewport", "content": "width=device-width"}], 
    title='Canopy',
    update_title='Cargando...',
    suppress_callback_exceptions=True
)

from apps import alerta_sobrecosto, alerta_transparencia, visor_datos, home_page, metodologia, concentracion, financiacion_campanias


server = app.server


# define page layout
app.layout = html.Div(
    [
        dcc.Location(id="url", refresh=False),
        # Column for user controls (SIDE BAR)
        dbc.Spinner(
            children=[
                html.Div(
                    id="content"
                ),
            ],
            size="lg", color="primary", type="border", 
            spinnerClassName='spinner',
            type=None,
            fullscreen=True,
        ),
    ]
)



# create callback for modifying page layout
@app.callback(
    Output("content", "children"),
    [Input("url", "pathname")])
def display_page(pathname):
    if pathname == "/":
        return home_page.layout
    if pathname == "/alerta-sobrecosto":
        return alerta_sobrecosto.layout
    # if not recognised, return home page
    return home_page.layout




if __name__ == "__main__":
    app.run_server(debug=True)
```

And the layouts (home_page, alerta_sobrecosto) could have any callback and the same error triggered.

**Comment by tcbegley** [comment]
Hi @melissadosm 

The spinner will show if any of the components it wraps is currently waiting for the result of a callback to update itself. Since you are wrapping the entire layout of each page with the spinner, unfortunately that will mean that all callbacks will trigger the callback. This is how loading components are implemented in Dash and it's not really in our control to change that behaviour (you would observe the same problems with `dcc.Loading` for example).

There is a workaround here which we actually use in the [dash-bootstrap-components component documentation](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/alert/), but it's a little involved. You can create multiple Dash apps, one for each page, and [embed them in a Flask app using `DispatcherMiddleware`](https://dash.plotly.com/integrating-dash#embedding-a-dash-app-within-an-existing-flask-app) so that the Flask app controls the navigation between pages. You can then add [some css](https://github.com/facultyai/dash-bootstrap-components/blob/master/docs/static/loading.css) to replace the `Loading...` on each Dash app with the spinner. It adds a fair bit of complexity though just for loading spinners.

**Comment by melissadosm** [comment]
Hi @tcbegley, thanks for the answer. However I don't see the same behavior with dcc.Loading, that's why I posted it as an issue. With dcc.Loading what happens is that the Loader does trigger, but then it returns to the same part of the page where the callback is, rather than returning to the beginning of the page.

**Comment by tcbegley** [comment]
Oh... my bad, sorry! I'll take another look and see if I can figure out why it breaks for `dbc.Spinner` but not `dcc.Loading`.

**Comment by tcbegley** [comment]
This should now be resolved as of dash-bootstrap-components 0.11.0.

Please update dash-bootstrap-components and let us know if you're still having problems.
